src/main/java/org/jenkinsci/plugins/workflow/cps/nodes/StepAtomNode.java

```diff
@@ -48,7 +48,7 @@
 
     public StepAtomNode(CpsFlowExecution exec, StepDescriptor d, FlowNode parent) {
         super(exec, exec.iotaStr(), parent);
-        this.descriptorId = d!=null ? d.getId() : null;
+        this.descriptorId = d!=null ? d.getId().intern() : null;
```

Comment:
That would be expected during _deserialization_, which you are already fixing elsewhere; but why here? Something is wrong if `Descriptor.getId()` varies from call to call.